SwiftUI/Model/DefaultKeys.swift

```diff
@@ -24,7 +24,7 @@ extension Defaults.Keys {
         "externalLinkLoadingPolicy", default: Brand.defaultExternalLinkPolicy
     )
     static let searchResultSnippetMode = Key<SearchResultSnippetMode>(
-        "searchResultSnippetMode", default: Brand.defaultSearchSnippetMode
+        "searchResultSnippetModeV2", default: Brand.defaultSearchSnippetMode
```

Comment:
Since we reduced the amount of options, we don't want former users to use their previous values in settings.
By changing the key, we successfully override former user defaults.

Reply:
I don't think this is the right approach. Instead, keep the same setting name, and create a compatibility and migration path for old values.

Reply:
Thank you. Good point.
I have updated it, and checked how the values are mapped for migrating users. 
Due to the underlying library and the enum values, it works now as expected:

| Former user value | Migrated to |
| ---- | ---- |
| disabled | disabled |
| first paragraph | matches |
| first sentence | matches |
| matches | matches |

Reply:
Great please add a clear comment in the code explaining what is going on (with link to this issue) and then a todo, to remove all of this in a few year

Reply:
I have added my explanation in code as well.